src/main/scala/org/hammerlab/guacamole/readsets/ReadSets.scala

```diff
@@ -0,0 +1,350 @@
+   *
+   * @param args parsed arguments
+   * @param sc spark context
+   * @param filters input filters to apply
+   */
+  def loadTumorNormalReads(args: TumorNormalReadsArgs,
+                           sc: SparkContext,
+                           filters: InputFilters): (ReadsRDD, ReadsRDD, ContigLengths) = {
+
+    val ReadSets(readsets, _, contigLengths) =
+      ReadSets(
+        sc,
+        Seq(args.tumorReads, args.normalReads),
+        filters,
+        !args.noSequenceDictionary,
+        ReadLoadingConfig(args)
+      )
+
+    (readsets(0), readsets(1), contigLengths)
+  }
+
+  /**
+    * Load reads from multiple files, merging their sequence dictionaries and verifying that they are consistent.
+    */
+  def apply(sc: SparkContext,
+            filenames: Seq[String],
+            filters: InputFilters = InputFilters.empty,
+            contigLengthsFromDictionary: Boolean = true,
+            config: ReadLoadingConfig = ReadLoadingConfig.default): ReadSets = {
+
+    val (readRDDs, sequenceDictionaries) = filenames.map(filename => load(filename, sc, filters, config)).unzip
+
+    val sequenceDictionary = mergeSequenceDictionaries(filenames, sequenceDictionaries)
+
+    val contigLengths: ContigLengths = {
+      if (contigLengthsFromDictionary) {
+        getContigLengthsFromSequenceDictionary(sequenceDictionary)
+      } else {
+        sc.union(readRDDs)
+          .flatMap(_.asMappedRead)
+          .map(read => read.referenceContig -> read.end)
+          .reduceByKey(math.max)
+          .collectAsMap()
+          .toMap
+      }
+    }
+
+    ReadSets(
+      readRDDs
+        .zip(filenames)
+        .map(ReadsRDD(_))
+        .toVector,
+      sequenceDictionary,
+      contigLengths
+    )
+  }
+
+
+  /**
+   * Given a filename and a spark context, return a pair (RDD, SequenceDictionary), where the first element is an RDD
+   * of Reads, and the second element is the Sequence Dictionary giving info (e.g. length) about the contigs in the BAM.
+   *
+   * @param filename name of file containing reads
+   * @param sc spark context
+   * @param filters filters to apply
+   * @return
+   */
+  private[readsets] def load(filename: String,
+                             sc: SparkContext,
+                             filters: InputFilters = InputFilters.empty,
+                             config: ReadLoadingConfig = ReadLoadingConfig.default): (RDD[Read], SequenceDictionary) = {
+
+    val (allReads, sequenceDictionary) =
+      if (filename.endsWith(".bam") || filename.endsWith(".sam"))
+        loadFromBAM(filename, sc, filters, config)
+      else
+        loadFromADAM(filename, sc, filters, config)
+
+    val reads = filterRDD(allReads, filters, sequenceDictionary)
+
+    (reads, sequenceDictionary)
+  }
+
+  /** Returns an RDD of Reads and SequenceDictionary from reads in BAM format **/
+  private def loadFromBAM(filename: String,
```

Comment:
Interesting. I think the right way to support what you're describing is to have an API that constructs a `ReadSets` from a list of filenames and corresponding list of filters. I think that would be worth adding.

Mimicking that functionality using a bunch of separate `load` calls should not be allowed because that directly undoes the small amount of worthwhile abstraction I've added here, namely the seqdict handling and APIs that treat a logical collection of read-sets collectively.